docs/lab/05/index.md

```diff
@@ -0,0 +1,209 @@
+---
+sidebar_position: 6
+description: Asynchronous Programming with Embassy
+slug: /lab/05
+---
+
+# 05 - Async/Await
+
+This lab will teach you the principles of asynchronous programming, and its application in Embassy-rs.
+
+
+## Resources
+
+TBD
+
+## Async functions
+
+Up to now, during the labs, we've seen that, in order to be able to do multiple different actions "at once", we would use *tasks*. We would let the main function run, while also doing another action seemingly "in parallel" inside of another task. 
+Let's take the following example: if we want to blink an LED every second while also waiting for a button press to do something else, we would need to spawn a new task in which we would wait for the button press, while blinking the LED in the main function. 
+
+When thinking of how exactly this works, you would probably think that the task is running on a separate *thread* than the main function. That would work in other cases, but not ours. Since only one thread can independently run per processor core, that means that, since we are using only one core of the RP2040 (which actually has only 2), we would only be able to run **one thread at a time**. And we don't have an OS to help us. So how exactly does the task wait for the button press in parallel with the LED blinking? 
+Short answer is: it doesn't. In reality, both functions runs asynchronously. 
+
+### Tasks
+
+Tasks in Embassy are what we call "asynchronous functions". Asynchronous functions are different from normal functions, in the sense that they allow asynchronous code execution. Let's take an example from the previous lab:
```

Comment:
Not every asynchronous function is a task, only the entry point is a task. Functions have to be annotated.

Reply:
I said that tasks are asynchronous functions. Is that a wrong statement?
Should I provide more information about what classifies as a task?

Reply:
My understanding of this is that "tasks are called asynchronous functions", which is incorrect. "A task in embassy is represented by an asynchronous function" or "has to be an asynchronous function" would be better.

Reply:
I understand the ambiguity of my wording now. I'll correct it.